### PR TITLE
Fix the uncorrect return value of Tensor.numpy()

### LINKED ISSUE #123494: [bug] Tensor.numpy() should return np.ndarray not Any
https://github.com/pytorch/pytorch/blob/3e8c64a637254b56dfcd8f5ddb3ffd201eed54d5/tools/pyi/gen_pyi.py#L1121

cc @ezyang @malfet @xuzhao9 @gramster

### PR DESCRIPTION
**Pull request by fffrog**
Fixes #123494

As the ISSUE stated.

### CODE REVIEW

**Review by XuehaiPan — APPROVED** [review]

File: torch/_C/__init__.pyi.in [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # mypy: disable-error-code="type-arg"
 
 import builtins
+import numpy
```
[the remark this answers is not in the document]
Comment (FFFrog) [reply torch/_C/__init__.pyi.in:5]: Done

**Comment by FFFrog** [comment]
> I would suggest to return numpy.ndarray[Any,Any] or numpy.typing.NDArray[Any] since np.ndarray is a just generic type.

Done

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**